File: vehiculo.java
/*
                  Universidad de las fuerzas armadas
                   Tecnologías de la información
Nombre: Juan Pablo Cano                             NRC: 13930
Fecha: 15-11-2023
                            
Autor: Juan Pablo Cano 
 */
package vehiculo_multa;

import java.util.Scanner;


public class vehiculo {
    //Atributos
     String marca;
     int velocidad;
     
    //Contructor

    public vehiculo() {
        this.marca = marca;
        this.velocidad = velocidad;
    }

     
    // Métodos 
    //Método para ingresar la velocidad del vehiculo
    
    public void ingresarVelocidad (){
        System.out.println("ingrese la velocidad del vehiculo: ".toUpperCase());
        Scanner leer = new Scanner(System.in);
        velocidad=leer.nextInt();
    }
    //Método para ingresar la marca del vehiculo
     public void ingresarMarca (){
        System.out.println("ingrese la marca del vehiculo: ".toUpperCase());
        Scanner leer = new Scanner(System.in);
        marca=leer.nextLine();
    }
    
    //Método para mostrar la velocidad del vehiculo
    public void mostrarVelocidad (){
        System.out.println("la velocidad del vehiculo es: ".toUpperCase()+this.velocidad);
    } 
    //Método para mostrar la marca del vehiculo
    public void mostrarMarca (){
        System.out.println("la marca del vehiculo es: ".toUpperCase()+this.marca);
    } 
    //Metodo para determinar el exceso de velocidad y si va a ser multado o no
    public void excesoVelocidad (){
        if (velocidad > 50){
            System.out.println("El vehiculo sera multado".toUpperCase());
        } 
        else{
            System.out.println("El vehiculo no sera multado".toUpperCase());
        }
    }
    
}

File: Primer Parcial/C++/Vehiculo.cpp
#include <iostream>
#include <string>

class Vehiculo {
private:
    std::string marca;
    int velocidad;

public:
    // Constructor
    Vehiculo() : marca(""), velocidad(0) {}

    // Método para ingresar la velocidad del vehículo
    void ingresarVelocidad() {
        std::cout << "Ingrese la velocidad del vehiculo: ";
        std::cin >> velocidad;
    }

    // Método para ingresar la marca del vehículo
    void ingresarMarca() {
        std::cout << "Ingrese la marca del vehiculo: ";
        std::getline(std::cin, marca);
    }

    // Método para mostrar la velocidad del vehículo
    void mostrarVelocidad() {
        std::cout << "La velocidad del vehiculo es: " << velocidad << std::endl;
    }

    // Método para mostrar la marca del vehículo
    void mostrarMarca() {
        std::cout << "La marca del vehiculo es: " << marca << std::endl;
    }

    // Método para determinar el exceso de velocidad y si va a ser multado o no
    void excesoVelocidad() {
        if (velocidad > 50) {
            std::cout << "El vehiculo sera multado" << std::endl;
        } else {
            std::cout << "El vehiculo no sera multado" << std::endl;
        }
    }
};

int main() {
    Vehiculo vehiculo;
    vehiculo.ingresarMarca();
    vehiculo.ingresarVelocidad();
    vehiculo.mostrarMarca();
    vehiculo.mostrarVelocidad();
    vehiculo.excesoVelocidad();

    return 0;
}
